common/djangoapps/xblock_django/user_service.py

```diff
@@ -35,6 +35,16 @@ def get_current_user(self):
         """
         return self._convert_django_user_to_xblock_user(self._django_user)
 
+    def get_external_user_id(self, type_name):
+        """
+        Returns an external user id of the given type.
+        Raises ValueError if the type doesn't exist.
+        """
+        external_id, _ = ExternalId.add_new_user_id(self._django_user, type_name)
+        if not external_id:
+            raise ValueError("External ID type: %s does not exist" % type_name)
```

Comment:
Just a comment that this is assuming the failure is due to the type not existing, which won't necessarily be the case.